fireapp/lib/presentation/volunteer_information/volunteer_information_viewmodel.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:fireapp/domain/models/volunteer_information.dart';
+import 'package:fireapp/domain/repository/volunteer_information_repository.dart';
+import 'package:fireapp/domain/request_state.dart';
+import 'package:fireapp/global/di.dart';
+import 'package:fireapp/presentation/fireapp_view_model.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:injectable/injectable.dart';
+import 'package:rxdart/rxdart.dart';
+
+@injectable
+class VolunteerInformationViewModel extends FireAppViewModel{
+  final VolunteerInformationRepository volunteerInformationRepository;
+
+  final BehaviorSubject<RequestState<VolunteerInformation>>
+    _volunteerInformation = BehaviorSubject.seeded(RequestState.initial());
+  Stream<RequestState<VolunteerInformation>> get volunteerInformation =>
+      _volunteerInformation.stream;
+
```

Comment:
You may want to add a second stream in here that would contain a better formatted availability model. You could also instead do that in the repository when mapping from the DTO

Reply:
what for?

Reply:
So you can actually show something useful for the availability